### PR TITLE
Fixing problem with invalid dates for subscription events

### PR DESCRIPTION
**Pull request by polis80cy**
`SubscriptionEvent.all()` would return a Promise with a validation error due to an erroneous `Date` type for `effective_date` and `event_date`.

image (1)

Changing data type from `Date` to `DateTime` works fine, is in line with the rest of the Py library handling the other endpoints, and it matches [doc](https://dev.chartmogul.com/reference/list-subscription-events).  

Added a new integration test.
Unit & integrations tests pass OK.

### CODE REVIEW

File: test/integration/test_fetch_subscription_events.py
Comment (briwa) [inline test/integration/test_fetch_subscription_events.py:9]: We should instead use the VCR cassette approach instead of using actual requests every time. Example: 
https://github.com/chartmogul/chartmogul-python/blob/c85832c56594f610621909dbf4a4ba1edc18abf8/test/integration/test_delete_invoice.py#L69-L70

[nit] And perhaps there's a better way to assert the result instead of printing it, just to make sure that they're all valid

**Review by briwa — COMMENTED** [review]
This PR should perhaps wait for #78 to be merged so that the CI could pass

**Review by pkopac — CHANGES_REQUESTED** [review]
We need to fix the data type, record a VCR cassette, remove the key and the `.DS_Store` file

**Review by sbrych — APPROVED** [review]
I have one question, it's non-blocking and LGTM otherwise ✈️

File: requirements.txt
Comment (polis80cy) [reply requirements.txt]: So the immediate next version is 2.0 which yes, it's breaking.

**Review by mynameispedro — APPROVED** [review]

**Review by briwa — APPROVED** [review]
Code coverage is still failing but it's unrelated to this PR, we can fix it later

File: requirements-test.txt
Comment (pkopac) [inline requirements-test.txt:5]: <= means including `2.0.0`, which is perhaps not the best as it will include the major version, but without any patch updates.

When protecting against major breaking upgrades you should use _exclusive lower than_ `<` and the version that's breaking.

Also, it appears to work, so you can actually do `< 3.0.0`

https://pip.pypa.io/en/stable/reference/requirement-specifiers/

File: requirements.txt
Comment (pkopac) [inline requirements.txt:6]: Why not `urllib3<2.0` ? It literally means _install anything lower but 2.0_

**Review by pkopac — COMMENTED** [review]
You can add `continue-on-error: true` to the codeclimate step to make it green like in #78. We need to figure out how to do it correctly later.

File: requirements.txt
Comment (polis80cy) [reply requirements.txt:6]: `1.26.15` is the last version before 2.0.0

**Review by pkopac — APPROVED** [review]

File: requirements.txt
Comment (pkopac) [reply requirements.txt:6]: The thinking goes: It's the latest now, but you don't know if there's a security bug uncovered later and a patch released as `1.26.16`. Anyway...

**Pull request merged**
